Assets/Scripts/Player/PlayerController.cs

```diff
@@ -62,14 +59,15 @@ public void SpawnBullet()
 
         public void DestroyPlayer()
         {
+            PlayerManager.Instance.playerSpawned -= InvokeEvents;
             GameUI.InstanceClass.Respawn(playerInput);
```

Comment:
why is gameUi re spawn the player? and there should be an event for this type of calling.